databroker/utils.py

```diff
@@ -414,3 +414,14 @@ def __iter__(self):
     def __contains__(self, k):
         # make sure checking 'in' does not trigger evaluation
         return k in self.__mapping
+
+    def add(self, *args, **kwargs):
+        dictionary = dict(*args, **kwargs)
+        wrap = self.__Wrapper
+        with self.__lock:
+            for key in dictionary:
```

Comment:
This might be a nice place to use set logic rather than nested for loops.

Reply:
updated